### PR TITLE
lightning: add function about write-then-read to external engine

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #45719

Problem Summary:

### What is changed and how it works?

we can use external storage, dataFiles and statFiles to construct an external.Engine, and call `createMergeIter`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/backend/external/util_test.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+	got, err := seekPropsOffsets(ctx, []byte("key2.5"), []string{file1, file2}, store)
+	require.NoError(t, err)
+	require.Equal(t, []uint64{10, 20}, got)
+	got, err = seekPropsOffsets(ctx, []byte("key3"), []string{file1, file2}, store)
+	require.NoError(t, err)
+	require.Equal(t, []uint64{30, 20}, got)
+	got, err = seekPropsOffsets(ctx, []byte("key0"), []string{file1, file2}, store)
+	require.NoError(t, err)
+	// TODO(lance6716): maybe return an error?
```
Comment (lance6716) [inline br/pkg/lightning/backend/external/util_test.go:97]: in the comment I wrote "the key at offset is less than or equal to the given start key." but this test we break the property. Should we return error? @tangenta @wjhuang2016

Comment (tangenta) [reply br/pkg/lightning/backend/external/util_test.go:97]: +1 for returning an error like 'start offsets not found' since this is not expected.

**Review by tangenta — COMMENTED** [review]
Rest LGTM

File: br/pkg/lightning/backend/external/engine.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	"encoding/hex"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/br/pkg/storage"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+)
+
+// Engine stored sorted key/value pairs in an external storage.
```
[the remark this answers is not in the document]
Comment (lance6716) [reply br/pkg/lightning/backend/external/engine.go:28]: Sorry I can't write a good comment in this PR. I need to learn more property when port more code in future.

File: br/pkg/lightning/backend/external/util_test.go (hunk 1 above)
Comment (lance6716) [reply br/pkg/lightning/backend/external/util_test.go]: https://github.com/pingcap/tidb/pull/46157/commits/5fa2dd2965c1627e7a58221d4934d472ada5d2a7

**Review by tangenta — APPROVED** [review]

**Pull request merged**